### PR TITLE
Fix/delete local ct only

### PR DESCRIPTION
**Pull request by mathieud**
## Context

Fixes a bug when deleting a CustomType. There was a mismatch between attribute missing and being undefined.

## The Solution

Update the types and code around to match the state given by the server

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [x] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [x] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/src/modules/availableCustomTypes/index.ts [hunk 1]
```diff
@@ -208,23 +211,48 @@ export const availableCustomTypesReducer: Reducer<
...
+        return {
+          ...state,
+          [ct.local.id]: deletedCT,
+        };
+      }
+      if (isNewCustomType(ct)) {
+        return omit(state, action.payload.customTypeId);
+      }
+
+      return state;
```
Comment (danigo99) [inline packages/slice-machine/src/modules/availableCustomTypes/index.ts:226]: [OPT] Do you think this case deserves a comment?

File: packages/slice-machine/src/modules/availableCustomTypes/types.ts [hunk 2]
```diff
@@ -11,25 +11,29 @@ export type SyncedCustomType = {
...
   ct: FrontEndCustomType
 ): ct is DeletedFrontEndCustomType => !isLocalCustomType(ct);
 
 export const isNewCustomType = (ct: FrontEndCustomType): ct is NewCustomType =>
-  isLocalCustomType(ct) && !("remote" in ct);
+  !("remote" in ct);
+
+export const isSyncedCustomType = (
+  ct: FrontEndCustomType
+): ct is SyncedCustomType => "remote" in ct && "local" in ct;
```
Comment (danigo99) [inline packages/slice-machine/src/modules/availableCustomTypes/types.ts:36]: ❓ Is this enough? We don't need to check that local & remote !== undefined?

File: packages/slice-machine/components/ReviewModal/index.tsx [hunk 3]
```diff
@@ -91,10 +92,12 @@ const ReviewModal: React.FunctionComponent = () => {
         }, 0)
       : 0;
 
-  const hasSliceWithinCustomType: boolean = customTypes.some((customType) =>
-    customType.local?.tabs.some(
-      (tab) => tab.sliceZone && tab.sliceZone?.value.length > 0
-    )
+  const hasSliceWithinCustomType: boolean = customTypes.some(
```
Comment (danigo99) [inline packages/slice-machine/components/ReviewModal/index.tsx]: ❓ Why do we need to explicitly define the type?

**Review by danigo99 — APPROVED** [review]

File: packages/slice-machine/src/modules/availableCustomTypes/index.ts (hunk 1 above)
Comment (mathieud) [reply packages/slice-machine/src/modules/availableCustomTypes/index.ts:226]: I'd say no, the types are explicit enough

File: packages/slice-machine/src/modules/availableCustomTypes/types.ts (hunk 2 above)
Comment (mathieud) [reply packages/slice-machine/src/modules/availableCustomTypes/types.ts:36]: remote being undefined is not a valid use case. It's either present and a custom type, or not defined at all

Comment (mathieud) [reply packages/slice-machine/src/modules/availableCustomTypes/types.ts:36]: See [here](https://github.com/prismicio/slice-machine/blob/964279b1e08823d215c38c0680569c9cd47e4afd/packages/slice-machine/src/modules/availableCustomTypes/types.ts#L4).
An attribute being undefined is what caused the crash we had before.

File: packages/slice-machine/components/ReviewModal/index.tsx (hunk 3 above)
Comment (mathieud) [reply packages/slice-machine/components/ReviewModal/index.tsx]: Didn't notice it. Updated.

**Review by danigo99 — APPROVED** [review]

**Pull request merged**